tests/ui/async-await/async-closures/no-borrow-from-env.rs

```diff
@@ -38,7 +38,10 @@ fn through_field_and_ref<'a>(x: &S<'a>) {
 
     let c = async move || { println!("{}", *x.0); };
     outlives::<'a>(c());
-    // outlives::<'a>(call_once(c)); // FIXME(async_closures): Figure out why this fails
+
+    // outlives::<'a>(call_once(c));
```

Comment:
Explanation for why this fails, since I understand it now, and it's totally expected.